tests/server/tso/manager_test.go

```diff
@@ -225,6 +226,44 @@ func (s *testPrioritySuite) TestAllocatorPriority(c *C) {
 	}
 }
 
+func (s *testPrioritySuite) TestNextLeaderGuarantee(c *C) {
+	dcLocationConfig := map[string]string{
+		"pd1": "dc-1",
+	}
+	serverNumber := len(dcLocationConfig)
+	cluster, err := tests.NewTestCluster(s.ctx, serverNumber, func(conf *config.Config, serverName string) {
+		conf.LocalTSO.EnableLocalTSO = true
+		conf.LocalTSO.DCLocation = dcLocationConfig[serverName]
+	})
+	defer cluster.Destroy()
+	c.Assert(err, IsNil)
+
+	c.Assert(failpoint.Enable("github.com/tikv/pd/server/tso/injectNextLeaderKey", "return(true)"), IsNil)
+	err = cluster.RunInitialServers()
+	c.Assert(err, IsNil)
+
+	cluster.WaitLeader()
+	// To speed up the test, we force to do the check
+	checkLeader := func() {
+		wg := sync.WaitGroup{}
+		for _, server := range cluster.GetServers() {
+			wg.Add(1)
+			go func(ser *tests.TestServer) {
+				ser.GetTSOAllocatorManager().ClusterDCLocationChecker()
+				wg.Done()
+			}(server)
+		}
+		wg.Wait()
+	}
```

Comment:
You can use `waitAllLeaders` here.

Reply:
I think `checkLeader` and `waitAllLeaders` are not equal here.